client/state/index.js

```diff
@@ -47,5 +47,8 @@ export function createReduxStore() {
 	) {
 		createStoreWithMiddleware = compose( createStoreWithMiddleware, window.devToolsExtension() );
 	}
-	return createStoreWithMiddleware( createStore )( reducer );
-};
+	if ( initialState === null ) {
+		initialState = {};
+	}
```

Comment:
tested and confirmed what you found, `null` doesn't trigger the default values. I also commented about this above, wondering if it's a meritable thing to guard against